server/routes/overview/mappers/prisonerSupportNeedsMapper.test.ts

```diff
@@ -0,0 +1,107 @@
+import moment from 'moment'
+import type { PrisonerSupportNeeds, HealthAndSupportNeeds, Neurodiversity } from 'viewModels'
+import { toPrisonerSupportNeeds } from './prisonerSupportNeedsMapper'
+
+describe('prisonerSupportNeedsMapper', () => {
+  it('should map to SupportNeeds', () => {
+    // Given
+    const learnerProfile = [
+      {
+        prn: 'G6123VU',
+        establishmentId: 'MDI',
+        establishmentName: 'MOORLAND (HMP & YOI)',
+        lddHealthProblem: 'Learner considers himself or herself to have a learning difficulty.',
+        languageStatus: 'Bilingual',
+        primaryLDDAndHealthProblem: 'Hearing impairment',
+        additionalLDDAndHealthProblems: [],
+      },
+      {
+        prn: 'G6123VU',
+        establishmentId: 'DNI',
+        establishmentName: 'DONCASTER (HMP)',
+        lddHealthProblem: 'Learner considers himself or herself to have a learning difficulty.',
+        languageStatus: 'Bilingual',
+        primaryLDDAndHealthProblem: 'Visual impairment',
+        additionalLDDAndHealthProblems: [
+          'Hearing impairment',
+          'Social and emotional difficulties',
+          'Mental health difficulty',
+        ],
+      },
+    ]
+    const learnerNeurodivergence = [
+      {
+        prn: 'G6123VU',
+        establishmentId: 'MDI',
+        establishmentName: 'MOORLAND (HMP & YOI)',
+        neurodivergenceSelfDeclared: ['Dyslexia'],
+        selfDeclaredDate: '2022-02-18',
+        neurodivergenceAssessed: ['ADHD'],
+        assessmentDate: '2022-05-18',
+        neurodivergenceSupport: ['Writing support'],
+        supportDate: '2022-02-18',
+      },
+      {
+        prn: 'G6123VU',
+        establishmentId: 'DNI',
+        establishmentName: 'DONCASTER (HMP)',
+        neurodivergenceSelfDeclared: [],
```

Comment:
I've asked if this will ever be null/undefined. I know this is only test data, but in the arrays below that we have `No identified...` type values, so I'm wondering if we can expect that kind of value as a default.

Reply:
I have defensively coded around the date fields and arrays (where relevant), expecting them to be null.